Implements blog functionality for visitors

Nandika-A opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

A new django app 'blog' is created.

models:
1. post: All details related to post including foreign key to user, like and dislike models.
2. like: Stores all the likes for every posts by every visitor.
3. dislike: Stores all the dislikes for every posts by every visitor.
4. comment: Stores comments.

views:

Functionality for List View, Detail View, Create, Update, Delete, like, dislike, comment on posts, and view the user's personal posts.

1. The functionality can be accessed by clicking on the blog tab on the Navigation bar. This leads to the list view of all the blogs.
2.  By clicking on the post title, users can access the detailed view of that post, where they can also like, dislike and comment if they are authenticated.
3.  If the user is that post's author, they can update or delete the post by clicking on the buttons provided. 
4. If a visitor likes/dislikes the post, the count is incremented/decremented. 
5. Comments can be viewed and new comments can be added by clicking on comment button, which directs to the comment page. 
6. Every page contains a side-bar, providing options to create new post, view personal posts and view profile if the user is authenticated, otherwise mentions them to login/signup.
7. 'Your posts' tab directs the user to the list view of all their posts.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

Fixes #65 

## Type of change

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
Adds a blog feature for visitors.

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!-- Ignore if not relevant for you -->

There are no errors after the implementation.

## Screenshots (if appropriate):

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.


**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

Nandika-A commented:
@kunal2812 I have implemented all the functionalities specified in the issue. Please check if any changes are required.

kunal2812 commented:
Remove auto-generated migration files and binary files in pycache folders and add those folders to .gitignore. Keep only those changes which are relevant to the issue resolved

Nandika-A commented:
@kunal2812 Sir I have made the required changes.

kunal2812 commented on blog/__pycache__/__init__.cpython-310.pyc:
Remove these as well

kunal2812 commented:
@Nandika-A The contents of the blog post are not visible when I click on the blog post.

![image](https://github.com/ContriHUB/Sheher/assets/59749539/f31381c6-cc52-422c-a95b-bba6befb4888)

Also please follow the website theme. Currently, it looks like a separate website altogether. Also when I comment on any post they should be visible below the post. As of now I can't see them anywhere.

Add the documentation changes as well.

```
python manage.py makemigrations blog
python manage.py migrate blog
```

Nandika-A commented:
@kunal2812 Sir I have made a separate page for viewing comments. I will move to the same page now.

Nandika-A commented:
@kunal2812 I have moved the comments to the post detail page and updated the readme.
![Screenshot (53)](https://github.com/ContriHUB/Sheher/assets/93462818/3534ce44-ad5f-482a-8aac-d42c37cfe433)
![Screenshot (54)](https://github.com/ContriHUB/Sheher/assets/93462818/19a19ed2-e12c-4080-823c-365aab63233c)

kunal2812 commented:
@Nandika-A Fix the theme of this page. This issue has bonus points.

Nandika-A commented:
@kunal2812 sir I included the website navbar in blog/base.html but it just included the HTML and not the CSS.

kunal2812 commented:
You can customise your existing navbar according to theme of the app

kunal2812 commented:
> @kunal2812 I have moved the comments to the post detail page and updated the readme. ![Screenshot (53)](https://user-images.githubusercontent.com/93462818/275141230-3534ce44-ad5f-482a-8aac-d42c37cfe433.png) ![Screenshot (54)](https://user-images.githubusercontent.com/93462818/275141238-19a19ed2-e12c-4080-823c-365aab63233c.png)

Why is it showing "None" in the comment section. It should show the username of the person who commented right ?

Nandika-A commented:
> > @kunal2812 I have moved the comments to the post detail page and updated the readme. ![Screenshot (53)](https://user-images.githubusercontent.com/93462818/275141230-3534ce44-ad5f-482a-8aac-d42c37cfe433.png) ![Screenshot (54)](https://user-images.githubusercontent.com/93462818/275141238-19a19ed2-e12c-4080-823c-365aab63233c.png)
> 
> Why is it showing "None" in the comment section. It should show the username of the person who commented right ?

sir in visitor/models, the username is set to none.

Nandika-A commented:
Should I use email instead of username here?

kunal2812 commented:
Yes just like the blogs

Nandika-A commented:
@kunal2812 Sir I have changed the theme.

Nandika-A commented:
@kunal2812 sir, are there any changes to be made?

kunal2812 commented:
@Nandika-A Ensure that the image is properly contained within the blog body, preventing it from flowing all over the page

![image](https://github.com/ContriHUB/Sheher/assets/59749539/66891c82-2f14-4d6e-aad4-746c510094c4)

Nandika-A commented:
@kunal2812 sir I have fixed the image size now.

kunal2812 commented:
![image](https://github.com/ContriHUB/Sheher/assets/59749539/e616487c-fe46-4fa4-96d3-52614559e512)

Login/Signup button isn't working

![image](https://github.com/ContriHUB/Sheher/assets/59749539/ebfaba88-d04f-49e2-9273-d7472f2ec551)

Timing is in UTC. Change to local time zone (The time was 6:50 pm when I took the screenshot).

Fix these issues. Rest everything is okay.

kunal2812 commented on media/images/post/Screenshot_21.png:
Is this file needed ?

kunal2812 commented on media/images/Visitors/cat_discord.png:
This one ?

kunal2812 commented on media/images/post/cat_discord_Qw7xVwD.png:
Remove unnecessary files.

Nandika-A commented:
I will delete these files

Nandika-A commented:
> ![image](https://user-images.githubusercontent.com/59749539/275328281-e616487c-fe46-4fa4-96d3-52614559e512.png)
> 
> Login/Signup button isn't working
> 
> ![image](https://user-images.githubusercontent.com/59749539/275328386-ebfaba88-d04f-49e2-9273-d7472f2ec551.png)
> 
> Timing is in UTC. Change to local time zone (The time was 6:50 pm when I took the screenshot).
> 
> Fix these issues. Rest everything is okay.

Sir, I have directly imported the navbar from navbar.html. I haven't changed anything in the code of login/signup.

Nandika-A commented:
@kunal2812 sir I have made the changes.
